### ISSUE TITLE
Enable embedding hyperlinks in TLegend

### ISSUE THREAD

**Issue by vepadulano** [root-project/root#16512]
### Explain what you would like to see improved and how.

I have a simple snippet which produces a plot (in PDF or SVG) where the text items in the legend are clickable hyperlinks:
```python
from matplotlib import pyplot as plt
plt.rcParams['svg.fonttype'] = 'none'
fig = plt.figure()
plt.scatter([1], [2], label="CERN")
plt.scatter([1], [2.25], label="ROOT")
leg = plt.legend()
for ta in leg.texts:
    t = ta.get_text()
    if t == "CERN":
        url = 'https://home.cern'
        ta.set_url(url)
    elif t == "ROOT":
        url = 'https://root.cern'
        ta.set_url(url)

fig.savefig('scatter.svg')
fig.savefig('scatter.pdf')
```

Can be run with `python repro.py` and saves both SVG and PDF files with the clickable links in the legend.

It would be nice to have the same feature also for TLegend. I have taken a look at [TLink](https://root.cern.ch/doc/master/classTLink.html) but despite what the docs hint at, that cannot be used to show hyperlinks, rather only links between different graphical components of the same canvas/browser. 

### ROOT version

Any

### Installation method

Any

### Operating system

Any

**Comment by dpiparo** [comment]
[scatter.pdf](https://github.com/user-attachments/files/17179775/scatter.pdf)
![scatter](https://github.com/user-attachments/assets/a0e848bc-0c65-4c98-845f-10d81764089f)

**Comment by couet** [comment]
After a discussion with @vepadulano it appears that it might be better to add a data member `fUrl` to `TAttText`, so that all the classes inheriting from it will benefit from this feature.

**Comment by linev** [comment]
I add preliminary support of URLs also in the JSROOT - see example link:

 https://jsroot.gsi.de/dev/?file=../files/tmp/latex_url.root&item=c1

It also works when export into PDF.

And I do not think that `TAttText` class is proper place for additional data member.
There are many derived classes like `TGaxis` or `TStyle` where such member will be useless - while there is no sense to provide URL for them.

Most straight solution - just introduce `TLatexUrl` which will have additional `TString fUrl` member.
And if necessary - introduce special `TLegendEntryUrl` for same purposes.

**Comment by linev** [comment]
@couet 

That also can be done - one can extend `TLatex` functionality, adding `#url` command:
```
auto l1 = new TLatex(0.5, 0.5, "Link on #url[https://root.cern]{root.cern} web site");
```
Using TLatex syntax, one can provide more attributes like colors or underline.

**Comment by couet** [comment]
@linev Yes that's also a nice possibility. Yes the TAttText approach might be not correct. You are right. What is the sense of TGaxis::SetUrl?

**Comment by linev** [comment]
> What is the sense of TGaxis::SetUrl?

This is exactly my point. I will provide preliminary implementation of `#url[]` in the latex parser of JSROOT.
I also will add demo in webcanvas tutorials.

**Comment by linev** [comment]
#16572 includes implementation of `#url[link]{label}` latex syntax for web canvas and provides demo for it.

Means in any place where `TLatex` used for rendering (histogram title, legend entry text, axis title) one can include
any web link. 

Unfortunately, such feature is very difficult to implement in classical graphics :(

**Comment by couet** [comment]
`#url` can be implemented in the classic TLatex, but that's a new keyword and it is not as easy as in web graphics. If it is really needed it can be done. However  having new funtionnalities in the new graphics, not present in the old one can also be a good way to motivate poeple to move toward the new graphics.

**Comment by couet** [comment]
Yes, it is a detail.  I am only saying that a new keyword is not mandatory if we follow the markdown syntax.

**Comment by linev** [comment]
>  if we follow the markdown syntax

I would not mix Markdown and TLatex.

TLatex is a single-line text, like axis or histogram title. It cannot provide Markdown features in this respect.

Markdown is more about page layout.

**Comment by couet** [comment]
Ok, up to you

**Comment by dpiparo** [comment]
Hi @couet and @linev : did we converge on a solution that is implementable and can address this issue?

**Comment by couet** [comment]
The web graphics version of TLatex has already this functionnality. See `tutorials/visualisation/webcanv/latex_url.cxx` . Do we want to have it also in the normal TLatex when a SVG file is genrated ? see the prototype I made (above) .

**Comment by linev** [comment]
Main difficulty is interactive graphics - it is fully new functionality which has to be implemented.
But if one starts implementing it - I propose to use `#url[https://root.cern]{root.cern}` syntax.

**Comment by couet** [comment]
Now when you run the example `tutorials/visualisation/webcanv/latex_url.cxx` with the standard graphics you see the full text with the url. Which is normal but no very nice. I am not saying we should implement the working URL in a the normal graphics pad but at least we can strip the URL from the text in order to display a nice text. In a second step we can go further and, for instance, implement it in TSVG as in the prototype above.

**Comment by couet** [comment]
Now implemented in TLatex. It works for the SVG output (TSVG) (see https://github.com/root-project/root/pull/21369). Some investagations showed it should be also possible to implement it for the PDF output (TPDF).

**Comment by dpiparo** [comment]
great. Is there an ETA for the implementation on PDFs? It will allow to close this item for good!

**Comment by couet** [comment]
I put a small PDF file here https://github.com/root-project/root/pull/21369 showing it can be done, but it needs to be implemented in TPDF. Some time is needed but hard to say how much

**Comment by couet** [comment]
https://github.com/root-project/root/pull/21369 has been merged. The TPDF implementation is more tricky and will come in another PR.

**Comment by couet** [comment]
TPDF implentation here: https://github.com/root-project/root/pull/21526

**Comment by dpiparo** [comment]
This is quite nice. Thanks!
The original ticket, https://github.com/root-project/root/issues/16512, also mentions png images with links. Is that something achievable?

**Comment by linev** [comment]
png image does not have any interactive elements. 
So any URL can be painted only as plain text on the image
